SU2_CFD/src/variables/CTurbVariable.cpp

```diff
@@ -35,8 +35,61 @@ CTurbVariable::CTurbVariable(unsigned long npoint, unsigned long ndim, unsigned
     turb_index.resize(nPoint) = su2double(1.0);
     intermittency.resize(nPoint) = su2double(1.0);
 
+    DES_LengthScale.resize(nPoint) = su2double(0.0);
+    SRSGridSize.resize(nPoint) = su2double(0.0);
+    
+    Vortex_Tilting.resize(nPoint);
+
    }
 
+
+void CTurbVariable::SetVortex_Tilting(unsigned long iPoint, CMatrixView<const su2double> PrimGrad_Flow,
+                                        const su2double* Vorticity, su2double LaminarViscosity) {
+
+  su2double Strain[3][3] = {{0,0,0}, {0,0,0}, {0,0,0}}, Omega, StrainDotVort[3], numVecVort[3];
+  su2double numerator, trace0, trace1, denominator;
+
+  AD::StartPreacc();
+  AD::SetPreaccIn(PrimGrad_Flow, nDim+1, nDim);
+  AD::SetPreaccIn(Vorticity, 3);
+  /*--- Eddy viscosity ---*/
+  AD::SetPreaccIn(muT(iPoint));
+  /*--- Laminar viscosity --- */
+  AD::SetPreaccIn(LaminarViscosity);
+
+  Strain[0][0] = PrimGrad_Flow[1][0];
+  Strain[1][0] = 0.5*(PrimGrad_Flow[2][0] + PrimGrad_Flow[1][1]);
+  Strain[0][1] = 0.5*(PrimGrad_Flow[1][1] + PrimGrad_Flow[2][0]);
+  Strain[1][1] = PrimGrad_Flow[2][1];
+  if (nDim == 3){
+    Strain[0][2] = 0.5*(PrimGrad_Flow[3][0] + PrimGrad_Flow[1][2]);
+    Strain[1][2] = 0.5*(PrimGrad_Flow[3][1] + PrimGrad_Flow[2][2]);
+    Strain[2][0] = 0.5*(PrimGrad_Flow[1][2] + PrimGrad_Flow[3][0]);
+    Strain[2][1] = 0.5*(PrimGrad_Flow[2][2] + PrimGrad_Flow[3][1]);
+    Strain[2][2] = PrimGrad_Flow[3][2];
```

Comment:
Don't we have a helper function for this somewhere? Maybe in the viscous fluxes?